coderd/prebuilds/api.go

```diff
@@ -0,0 +1,23 @@
+package prebuilds
+
+import (
+	"context"
+
+	"github.com/coder/coder/v2/coderd/database"
+)
+
+type ReconciliationOrchestrator interface {
+	Reconciler
+
+	RunLoop(ctx context.Context)
+	Stop(ctx context.Context, cause error)
+}
+
+type Reconciler interface {
```

Comment:
Looks like these API too low-level, I'd rather expose methods such as:
- ReconcileAll()
- ReconcilePreset()

But `SnapshotState` & `DetermineActions (CalcState)` are used in `MetricsCollector` - so probably keep it.

Reply:
I agree with this.

Reply:
Maybe make it an internal interface and internal test?

Reply:
I added a detailed comments, explaining how this API can be used, and why it may sense to expose all these API.

Reply:
I'd prefer to keep this "lower-level" interface unexported for now.

Reply:
@johnstcn I also wanted to do it, but:
- they will be used in `MetricsCollector`: https://github.com/coder/coder/pull/16523
- they are used for tests

So removing them means:
- I have to find workaround for tests
- later we have to find workaround for metrics collector

Reply:
There is no need for this interface to be exported for the metrics collector to work correctly, since it lives in the same package as the `StoreReconciler`.

Tests can be inside the same package if they need to access unexported functions. Also, you don't need to export an interface to call exported functions on, say, the `StoreReconciler`.

Reply:
Can we do this refactor in a follow-up? Let's try keep the momentum on this PR.

Reply:
I don't agree that "momentum" should be a reason to check in code that doesn't meet our standards. What is to be gained by fixing it later rather than now? IMO it just makes it more likely that we don't fix the issue at all.